### ISSUE TITLE
Slice assert_equiv error

### ISSUE THREAD

**Issue by ehsantn** [IntelLabs/numba#48]
The program below fails because the size of the slice is the same as the whole array.

```python
import numpy as np
import numba

@numba.njit(parallel=True)
def f(m, A):
    B = np.zeros(m)
    n = len(A)
    B[-n:] = A
    return B

print(f(10, np.ones(10)))
```
```
AssertionError: Sizes of B, A do not match on tmp_tests/slice_test.py (9)
```

**Comment by ninegua** [comment]
Good catch! Should now be fixed by 0929af4a63cacbf69088913052bace2e8074f5be

**Comment by ehsantn** [comment]
Great, thanks!
